src/include/CacheManager.hpp

```diff
@@ -0,0 +1,61 @@
+/**   
+ * MIT License
+ *
+ * Copyright (c) 2021 Tsuhao Chang
+ * 
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+#ifndef CACHE_MANAGER_H_
+#define CACHE_MANAGER_H_
+
+#include <string>
+#include <functional>
+#include <map>
+#include <mutex>
+
+#include "CacheContent.hpp"
+
+/**
+ * 將檔案內容載入記憶體
+ * 內部保存一個map，將key(應為request url)對應到一個CacheContent
+ * Example: 
+ * CacheManager cm = new CacheManager();
+ * cm.get("some/file.txt", someCallback);
+ */
+class CacheManager // singleton
+{
+public:
+    CacheManager();
+    CacheManager(const CacheManager& other) = delete;
+    CacheManager& operator=(const CacheManager&) = delete;
+    ~CacheManager();
+    const CacheManager &getInstance();
```

Comment:
Return a pointer?

Reply:
If you return a const reference, caller will not be able to call non-const method.

Reply:
fixed